wifi.go

```diff
@@ -0,0 +1,134 @@
+package keylight
+
+import (
+	"bytes"
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// A WiFiInfo contains the information neccessary for a Key Light to connect
+// to a wireless network.
+type WiFiInfo struct {
+	SSID         string `json:"ssid,omitempty"`
+	Passphrase   string `json:"passphrase,omitempty"`
+	SecurityType int    `json:"securityType,omitempty"`
+}
+
+const (
+	SecurityTypeNone int = 0
+	SecurityTypeWEP  int = 1
+	SecurityTypePSK  int = 2
+)
+
+// SetWiFiInfo encrypts and updates a Key Light's WiFi configuration.
+func (c *Client) SetWiFiInfo(ctx context.Context, wifi *WiFiInfo, device *Device) error {
+	b, err := json.Marshal(wifi)
+	if err != nil {
+		return err
+	}
+
+	blen := len(b)
```

Comment:
I can turn this into (un)pad functions if desired.

Reply:
Nah seems okay, just leave a comment explaining why you're doing this since a lot of folks aren't familiar with crypto.